rfcs/2021-06-02-7709-helm-deprecate-custom-config-keys.md

```diff
@@ -0,0 +1,97 @@
+# RFC 7709 - 2021-06-02 - Helm: Deprecate "custom" configuration keys
+
+Today the templates that user's leverage to generate their Vector config are separate from the ones used to generate the "default" Vector configuration.
+While these are not very complex in their functionality they do duplicate much of the same logic.
+
+## Scope
+
+This RFC will cover only the existing "default" `sinks` and `sources` defined in our Helm charts today.
+
+vector-agent:
+
+- `kubernetes_logs`
+- `vector`
+- `internal_metrics`
+- `host_metrics`
+- `prometheus_exporter`
+
+vector-aggregator:
+
+- `vector`
+- `internal_metrics`
+- `prometheus_exporter`
+
+## Motivation
+
+We're duplicating the templating used to generate Vector's configuration file in Helm based deployments. While I was working on transitioning the
+currently TOML config to YAML realized the configuration building had several unexpected complications due to the separation between Vector's provided
+"default" configurations (listed in the Scope section) and a user provided configuration located under the `sources`, `transforms`, and `sinks` keys.
```

Comment:
This paragraph reads a little weird, mostly in part because of the narration about how you got to the problem rather than just stating the problem plainly.

Consider how something like this reads instead:

> Our current Helm chart for configuring Vector in Kubernetes provides shortcuts for configuring common 
> sources and sinks, while also allowing users to specify their configuration in the more traditional "direct" 
> fashion, similar to configuring Vector when running on a virtual machine.  These two approaches being 
> used simultaneously can lead to issues, and general confusion, about what the correct way to configure Vector is. <maybe add additional pitfalls of this approach here>

Reply:
Ah yeah, I tend to have a very "train of thought" writing style.